dandi/tests/test_download.py

```diff
@@ -703,3 +703,20 @@ def test_progress_combiner(
     for path, status in inputs:
         outputs.extend(pc.feed(path, status))
     assert outputs == expected
+
+
+@pytest.mark.parametrize(
+    "url_path,asset_path,target",
+    [
+        ("", "foo/bar", "foo/bar"),
+        ("fo", "foo/bar", "foo/bar"),
+        ("foo", "foo/bar", "foo/bar"),
+        ("foo/", "foo/bar", "foo/bar"),
```

Comment:
note: so this one shows that it is a bit different from how I thought it in https://github.com/dandi/dandi-cli/issues/1025#issuecomment-1143964516 where it would have been
```suggestion
        ("foo/", "foo/bar", "bar"),
```
because of  trailing / in the `url_path`, but code rstrips it thus making it "consistent" between the two forms